src/server/rdb_load.cc

```diff
@@ -1585,16 +1585,20 @@ auto RdbLoaderBase::ReadHMap(int rdbtype) -> io::Result<OpaqueObj> {
 
   unique_ptr<LoadTrace> load_trace(new LoadTrace);
 
-  if (rdbtype == RDB_TYPE_HASH) {
-    len *= 2;
-  } else {
-    DCHECK_EQ(rdbtype, RDB_TYPE_HASH_WITH_EXPIRY);
-    len *= 3;
+  size_t increment = 2;
+  if (rdbtype == RDB_TYPE_HASH_WITH_EXPIRY) {
+    increment = 3;
   }
+  len *= increment;
 
   load_trace->arr.resize((len + kMaxBlobLen - 1) / kMaxBlobLen);
   for (size_t i = 0; i < load_trace->arr.size(); ++i) {
-    size_t n = std::min<size_t>(len, kMaxBlobLen);
+    size_t n = len;
+    if (n > kMaxBlobLen) {
+      // Ensure n is a multiple of increment.
+      n = kMaxBlobLen - (kMaxBlobLen % increment);
```

Comment:
Lets define `kMaxBlobLen` be a multiple of 6 with a comment why it's needed.

Reply:
sure updated